core/commands/up_command.rb

```diff
@@ -77,8 +77,8 @@ def generate_vagrant_run_flags(provider)
   # @param node [String] name of the node to get status from.
   # @param logger [Out] logger to log information to
   # @return [Boolean]
-  def node_running?(node, logger)
-    result = run_command("vagrant status #{node}", {}, logger)
+  def self.node_running?(node, logger)
```

Comment:
You must not leave this method in the UpCommand. Two commands should depend on each other only if one command executes another one, they must not depend upon each other on the code level.

You should instead create a service. For example it could be named VagrantCommands and provide methods for the caller.